src/EfficientDynamoDb/Internal/Operations/ExecuteStatement/ExecuteStatementRequestHttpContent.cs

```diff
@@ -0,0 +1,57 @@
+﻿using EfficientDynamoDb.Converters;
+using EfficientDynamoDb.Internal.Extensions;
+using EfficientDynamoDb.Internal.Operations.Shared;
+using EfficientDynamoDb.Operations.ExecuteStatement;
+using EfficientDynamoDb.Operations.Shared;
+using System.Threading.Tasks;
+
+namespace EfficientDynamoDb.Internal.Operations.ExecuteStatement
+{
+    internal class ExecuteStatementRequestHttpContent : DynamoDbHttpContent
+    {
+        private readonly ExecuteStatementRequest _request;
+
+        public ExecuteStatementRequestHttpContent(ExecuteStatementRequest executeStatementRequest) : base("DynamoDB_20120810.ExecuteStatement")
+        {
+            _request = executeStatementRequest;
+        }
+
+        protected override ValueTask WriteDataAsync(DdbWriter writer)
+        {
+            var json = writer.JsonWriter;
+            json.WriteStartObject();
+
+            json.WritePropertyName("Statement");
+            json.WriteStringValue(_request.Statement);
+
+            json.WritePropertyName("Parameters");
+            json.WriteStartArray();
+            foreach (var parameter in _request.Parameters)
+            {
+                parameter.Write(json);
+            }
+            json.WriteEndArray();
```

Comment:
`_request.Parameters` can be empty. In this case the DDB request shouldn't include the "Parameters" block at all because DDB fails when provided with empty parameters array. Wrapping this whole block with `if (_request.Parameters.Count > 0)` fixes the issue.

The same applies for transact and batch operations.